### ISSUE TITLE
[B5] Crash changing wand foci in gauntlet

### ISSUE THREAD

**Issue by KryptonCaptain** [Azanor/thaumcraft-beta#283]
```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  LoadingPlugin (Quark-r1.1-70.jar)
Contact their authors BEFORE contacting forge

// Uh... Did I do that?

Time: 14/02/18 10:44 PM
Description: Unexpected error

java.lang.NullPointerException: Unexpected error
	at thaumcraft.api.casters.FocusCore.getSortingHelper(FocusCore.java:238)
	at thaumcraft.common.items.casters.ItemFocus.getSortingHelper(ItemFocus.java:68)
	at thaumcraft.client.lib.events.WandRenderingHandler.handleFociRadial(WandRenderingHandler.java:78)
	at thaumcraft.client.lib.events.RenderEventHandler.renderOverlay(RenderEventHandler.java:381)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_144_RenderEventHandler_renderOverlay_RenderGameOverlayEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
	at net.minecraftforge.client.GuiIngameForge.renderHUDText(GuiIngameForge.java:694)
	at net.minecraftforge.client.GuiIngameForge.func_175180_a(GuiIngameForge.java:171)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1072)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1076)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:196)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:231)
	at org.multimc.EntryPoint.listen(EntryPoint.java:143)
	at org.multimc.EntryPoint.main(EntryPoint.java:34)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Client thread
Stacktrace:
	at thaumcraft.api.casters.FocusCore.getSortingHelper(FocusCore.java:238)
	at thaumcraft.common.items.casters.ItemFocus.getSortingHelper(ItemFocus.java:68)
	at thaumcraft.client.lib.events.WandRenderingHandler.handleFociRadial(WandRenderingHandler.java:78)
	at thaumcraft.client.lib.events.RenderEventHandler.renderOverlay(RenderEventHandler.java:381)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_144_RenderEventHandler_renderOverlay_RenderGameOverlayEvent.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
	at net.minecraftforge.client.GuiIngameForge.renderHUDText(GuiIngameForge.java:694)
	at net.minecraftforge.client.GuiIngameForge.func_175180_a(GuiIngameForge.java:171)

```

**Comment by Danymaddox** [comment]
What did you do? What type of focus where you using?

**Comment by Danymaddox** [comment]
Well, I have this crash every time I try to equip a focus made with "Architect" + "Exchange". 

No extra mods installed, only Baubles and Thaumcraft Beta5. 
Forge: 12.8.3.2511

Crash log:

> ---- Minecraft Crash Report ----
> // Why is it breaking :(
> 
> Time: 2/14/18 3:55 PM
> Description: Unexpected error
> 
> java.lang.NullPointerException: Unexpected error
> 	at thaumcraft.common.items.casters.ItemCaster.getPickedBlock(ItemCaster.java:559)
> 	at thaumcraft.client.lib.events.HudHandler.renderCastingWandHud(HudHandler.java:598)
> 	at thaumcraft.client.lib.events.HudHandler.renderHuds(HudHandler.java:84)
> 	at thaumcraft.client.lib.events.RenderEventHandler.renderTick(RenderEventHandler.java:313)
> 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_70_RenderEventHandler_renderTick_RenderTickEvent.invoke(.dynamic)
> 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
> 	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
> 	at net.minecraftforge.fml.common.FMLCommonHandler.onRenderTickEnd(FMLCommonHandler.java:355)
> 	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1078)
> 	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
> 	at net.minecraft.client.main.Main.main(SourceFile:124)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:483)
> 	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
> 	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
> 
> 
> A detailed walkthrough of the error, its code path and all known details is as follows:
> ---------------------------------------------------------------------------------------
> 
> -- Head --
> Thread: Client thread
> Stacktrace:
> 	at thaumcraft.common.items.casters.ItemCaster.getPickedBlock(ItemCaster.java:559)
> 	at thaumcraft.client.lib.events.HudHandler.renderCastingWandHud(HudHandler.java:598)
> 	at thaumcraft.client.lib.events.HudHandler.renderHuds(HudHandler.java:84)
> 	at thaumcraft.client.lib.events.RenderEventHandler.renderTick(RenderEventHandler.java:313)
> 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_70_RenderEventHandler_renderTick_RenderTickEvent.invoke(.dynamic)
> 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
> 	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
> 
> -- Affected level --
> Details:
> 	Level name: MpServer
> 	All players: 1 total; [EntityPlayerSP['Danymaddox'/166, l='MpServer', x=848.05, y=5.00, z=1174.27]]
> 	Chunk stats: MultiplayerChunkCache: 622, 622
> 	Level seed: 0
> 	Level generator: ID 01 - flat, ver 0. Features enabled: false
> 	Level generator options: 
> 	Level spawn location: World: (859,4,1190), Chunk: (at 11,0,6 in 53,74; contains blocks 848,0,1184 to 863,255,1199), Region: (1,2; contains chunks 32,64 to 63,95, blocks 512,0,1024 to 1023,255,1535)
> 	Level time: 66892 game time, 52515 day time
> 	Level dimension: 0
> 	Level storage version: 0x00000 - Unknown?
> 	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
> 	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: false
> 	Forced entities: 51 total; [EntitySlime['Slime'/103171, l='MpServer', x=887.68, y=4.00, z=1165.08], EntitySlime['Slime'/141, l='MpServer', x=891.17, y=4.00, z=1166.78], EntitySlime['Slime'/143, l='MpServer', x=914.82, y=4.00, z=1252.49], EntitySheep['Sheep'/20, l='MpServer', x=772.21, y=4.00, z=1237.74], EntitySlime['Slime'/161434, l='MpServer', x=913.62, y=4.75, z=1138.28], EntitySheep['Sheep'/31, l='MpServer', x=770.13, y=4.00, z=1128.76], EntitySheep['Sheep'/32, l='MpServer', x=774.17, y=4.00, z=1128.66], EntitySlime['Slime'/161, l='MpServer', x=911.65, y=5.17, z=1183.97], EntitySheep['Sheep'/34, l='MpServer', x=771.28, y=4.00, z=1142.96], EntityItem['item.item.slimeball'/165, l='MpServer', x=831.30, y=4.00, z=1188.48], EntityPig['Pig'/37, l='MpServer', x=769.91, y=4.00, z=1241.57], EntitySlime['Slime'/187303, l='MpServer', x=832.37, y=4.00, z=1206.53], EntitySheep['Sheep'/38, l='MpServer', x=773.24, y=4.00, z=1240.67], EntitySlime['Slime'/203311, l='MpServer', x=798.95, y=5.17, z=1116.11], EntitySlime['Slime'/172079, l='MpServer', x=793.27, y=5.25, z=1112.51], EntityItem['item.item.arrow'/47, l='MpServer', x=798.14, y=4.00, z=1174.60], EntityItem['item.item.bone'/48, l='MpServer', x=797.61, y=4.00, z=1174.84], EntitySlime['Slime'/75826, l='MpServer', x=916.47, y=5.17, z=1131.38], EntityItem['item.item.arrow'/52, l='MpServer', x=805.78, y=4.00, z=1148.23], EntityItem['item.item.arrow'/53, l='MpServer', x=806.05, y=4.00, z=1181.25], EntityPlayerSP['Danymaddox'/166, l='MpServer', x=848.05, y=5.00, z=1174.27], EntityItem['item.item.rottenFlesh'/57, l='MpServer', x=826.25, y=4.00, z=1133.94], EntityItem['item.item.rottenFlesh'/58, l='MpServer', x=827.85, y=4.00, z=1125.64], EntityItem['item.item.rottenFlesh'/59, l='MpServer', x=823.08, y=4.00, z=1133.32], EntityMinecartEmpty['entity.MinecartRideable.name'/60, l='MpServer', x=829.55, y=4.06, z=1122.45], EntityTurretCrossbowAdvanced['Advanced Automated Crossbow'/61, l='MpServer', x=829.55, y=4.06, z=1122.45], EntityMinecartEmpty['entity.MinecartRideable.name'/62, l='MpServer', x=829.50, y=4.06, z=1130.40], EntityTurretCrossbowAdvanced['Advanced Automated Crossbow'/63, l='MpServer', x=829.50, y=4.06, z=1130.40], EntityMinecartEmpty['entity.MinecartRideable.name'/65, l='MpServer', x=829.50, y=4.06, z=1153.20], EntityTurretCrossbowAdvanced['Advanced Automated Crossbow'/66, l='MpServer', x=829.50, y=4.06, z=1153.20], EntitySlime['Slime'/69, l='MpServer', x=831.28, y=4.00, z=1201.86], EntityItem['item.item.slimeball'/74, l='MpServer', x=841.23, y=4.00, z=1164.34], EntityItem['item.item.string'/75, l='MpServer', x=838.67, y=4.00, z=1159.36], EntityTippedArrow['Arrow'/76, l='MpServer', x=837.28, y=4.00, z=1168.64], EntityItem['item.item.slimeball'/79, l='MpServer', x=834.75, y=4.00, z=1190.00], EntitySlime['Slime'/31697, l='MpServer', x=907.30, y=4.00, z=1112.19], EntityItem['item.item.arrow'/84, l='MpServer', x=861.61, y=4.00, z=1114.42], EntityItem['item.item.bone'/85, l='MpServer', x=861.12, y=4.00, z=1115.76], EntityItem['item.item.sulphur'/87, l='MpServer', x=856.64, y=4.00, z=1167.29], EntityItem['item.item.sulphur'/88, l='MpServer', x=859.51, y=4.00, z=1171.78], EntitySlime['Slime'/89, l='MpServer', x=842.51, y=4.08, z=1219.85], EntitySlime['Slime'/90, l='MpServer', x=868.47, y=4.00, z=1219.46], EntitySlime['Slime'/192478, l='MpServer', x=794.01, y=4.00, z=1104.85], EntitySlime['Slime'/95205, l='MpServer', x=770.44, y=4.04, z=1175.96], EntitySlime['Slime'/188906, l='MpServer', x=924.33, y=4.00, z=1115.17], EntitySlime['Slime'/119528, l='MpServer', x=792.32, y=4.17, z=1109.07], EntitySlime['Slime'/172524, l='MpServer', x=923.73, y=4.00, z=1108.59], EntitySlime['Slime'/119, l='MpServer', x=881.55, y=4.00, z=1163.78], EntitySlime['Slime'/120, l='MpServer', x=909.86, y=4.00, z=1139.57], EntitySlime['Slime'/185208, l='MpServer', x=795.62, y=5.08, z=1104.95], EntitySlime['Slime'/188921, l='MpServer', x=844.83, y=5.17, z=1218.28]]
> 	Retry entities: 0 total; []
> 	Server brand: fml,forge
> 	Server type: Integrated singleplayer server
> Stacktrace:
> 	at net.minecraft.client.multiplayer.WorldClient.func_72914_a(WorldClient.java:415)
> 	at net.minecraft.client.Minecraft.func_71396_d(Minecraft.java:2660)
> 	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:400)
> 	at net.minecraft.client.main.Main.main(SourceFile:124)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
> 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.lang.reflect.Method.invoke(Method.java:483)
> 	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
> 	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
> 
> -- System Details --
> Details:
> 	Minecraft Version: 1.10.2
> 	Operating System: Windows 10 (amd64) version 10.0
> 	Java Version: 1.8.0_25, Oracle Corporation
> 	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
> 	Memory: 831950680 bytes (793 MB) / 2393899008 bytes (2283 MB) up to 5726797824 bytes (5461 MB)
> 	JVM Flags: 4 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx6144m -Xms256m -XX:PermSize=256m
> 	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
> 	FML: MCP 9.32 Powered by Forge 12.18.3.2511 5 mods loaded, 5 mods active
> 	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
> 	UCHIJAAAA	mcp{9.19} [Minecraft Coder Pack] (minecraft.jar) 
> 	UCHIJAAAA	FML{8.0.99.99} [Forge Mod Loader] (forge-1.10.2-12.18.3.2511.jar) 
> 	UCHIJAAAA	Forge{12.18.3.2511} [Minecraft Forge] (forge-1.10.2-12.18.3.2511.jar) 
> 	UCHIJAAAA	Baubles{1.3.13} [Baubles] (Baubles-1.10.2-1.3.13.jar) 
> 	UCHIJAAAA	thaumcraft{6.0.BETA5} [Thaumcraft] (Thaumcraft-1.10.2-6.0.BETA5.jar) 
> 	Loaded coremods (and transformers): 
> 	GL info: ' Vendor: 'NVIDIA Corporation' Version: '4.6.0 NVIDIA 388.13' Renderer: 'GeForce GTX 750/PCIe/SSE2'
> 	Launched Version: forge-12.18.3.2511
> 	LWJGL: 2.9.4
> 	OpenGL: GeForce GTX 750/PCIe/SSE2 GL version 4.6.0 NVIDIA 388.13, NVIDIA Corporation
> 	GL Caps: Using GL 1.3 multitexturing.
> Using GL 1.3 texture combiners.
> Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
> Shaders are available because OpenGL 2.1 is supported.
> VBOs are available because OpenGL 1.5 is supported.
> 
> 	Using VBOs: No
> 	Is Modded: Definitely; Client brand changed to 'fml,forge'
> 	Type: Client (map_client.txt)
> 	Resource Packs: 
> 	Current Language: English (US)
> 	Profiler Position: N/A (disabled)
> 	CPU: 4x Intel(R) Core(TM) i5-4690 CPU @ 3.50GHz

**Comment by Artemon90** [comment]
Same problem(
